app/domain/show-domain/src/main/java/org/example/usecase/ArtistUseCase.java

```diff
@@ -104,14 +84,40 @@ public List<Artist> findOrCreateArtistBySpotifyId(List<String> spotifyIds) {
             .map(ArtistSearchPortParam::toArtist)
             .toList();
 
-        artistCreatePort.createArtist(
-            "createArtist",
-            ArtistCreatePortRequest.from(response.artists(), newArtists)
-        );
+        saveArtists(response.artists(), newArtists);
 
         return Stream.concat(existArtists.stream(), newArtists.stream()).toList();
     }
 
+    public void saveArtists(List<ArtistSearchPortParam> params, List<Artist> newArtists) {
+        List<ArtistGenreDomainRequest> artistGenres = IntStream.range(0, params.size())
+            .mapToObj(i -> params.get(i).toDomainRequest(newArtists.get(i).getId()))
+            .toList();
+
+        List<Genre> existGenres = genreRepository.findAllByNameIn(
+            artistGenres.stream()
+                .map(ArtistGenreDomainRequest::genreName)
+                .toList()
+        );
+
+        for (ArtistGenreDomainRequest artistGenre : artistGenres) {
+            Genre genre = existGenres.stream()
+                .filter(exsistGenre -> exsistGenre.getName().equals(artistGenre.genreName()))
+                .findFirst()
+                .orElseThrow(NoSuchElementException::new);
+
+            Artist newArtist = artistGenre.toArtist();
+
+            transactionTemplate.execute(
+                status -> {
+                    artistRepository.save(newArtist);
+                    artistGenreRepository.save(newArtist.toArtistGenre(genre.getId()));
+                    return null;
+                }
+            );
```

Comment:
리뷰 포인트 3

execute과 함께 리턴값이 필요 없어서 null을 리턴하는 경우, 다음과 같이 수정하여도 같은 동작을 할 것 같아요!
```suggestion
            transactionTemplate.executeWithoutResult(
                status -> {
                    artistRepository.save(newArtist);
                    artistGenreRepository.save(newArtist.toArtistGenre(genre.getId()));
                }
            ); 
```